### DISCUSSION TITLE
Footer with custom information

### DISCUSSION THREAD

**Discussion by goldenratio** [withastro/starlight#2163] [Q&A] [answered]
I want to create a footer and attach it to all pages.
- I am auto generating .mdx from swagger API schema
- I want to show the schema version in the footer

What's the ideal way to do it? Thanks for your help in advance

**Comment by lorenzolewis** [comment, marked as answer]
Hi @goldenratio ! What you're most likely looking for are [Component Overrides](https://starlight.astro.build/guides/overriding-components/). You can [wrap one of the components](https://starlight.astro.build/guides/overriding-components/#reuse-a-built-in-component) there and then add the footer content that you would like.

Most likely the [Footer](https://starlight.astro.build/reference/overrides/#footer-1) would be the component you are looking for.

For a more interactive version of this to see exactly what component is used in a specific location on a page you can use [Starlight Overrides Map](https://starlight-overrides-map.netlify.app/example/), an interactive community reference created by @HiDeoo

**Reply by lorenzolewis** [reply]
Almost forgot about your specific ask on adding a schema version. How and where is that data stored right now? Is it in the front matter of some MDX? Is it something that could be derived from the filename? Depending on that there are a few different places you could read that are available at build-time to render it appropriately.
